lib/robots/viomi/ViomiValetudoRobot.js

```diff
@@ -211,23 +193,38 @@ class ViomiValetudoRobot extends MiioValetudoRobot {
                     key = msg.params.suffix;
                 }
 
-                let mapUploadUrls = [];
-                for (let i = 0; i < 4; i++) {
-                    let url = this.mapUploadUrlPrefix +
-                        "/api/miio/map_upload_handler?" +
-                        process.hrtime().toString().replace(/,/g, "") +
-                        "&suffix=" + key;
-                    mapUploadUrls.push({"url": url, "method": "PUT"});
+                let result = {ok: true};
+
+                let url = this.mapUploadUrlPrefix + "/api/miio/map_upload_handler?" +
```

Comment:
Roborock uses the same commands but with a different looking payload it seems?

https://github.com/Hypfer/Valetudo/blob/07a1c7ad78e3593b90e12ba0bb8286ed0c4f58e1/lib/robots/roborock/RoborockValetudoRobot.js#L102-L115

But dreame seems to be the same
https://github.com/Hypfer/Valetudo/blob/07a1c7ad78e3593b90e12ba0bb8286ed0c4f58e1/lib/robots/dreame/DreameValetudoRobot.js#L50-L80

Can we refactor this?

Reply:
Also, I wonder if roborock would also work with the same structure. Considering that two different vendors use that, it might be standard xiaomi stuff?

Reply:
That's possible. This is what I get on viomi with the stock app:

vac → cloud: `{"id":295,"method":"_sync.gen_presigned_url","params":{"suffix":"bin"}}`
cloud → vac: 
```json
{"id":295,"result":{"bin":
{"url":
"http://awsde0.fds.api.xiaomi.com/viomi-device/2021/04/10/####/######.bin?Expires=1618049385000&GalaxyAccessKeyId=#######&Signature=zH0kgTPE1XUQPzR
74Vv2vsmAbtI=",
"obj_name":"2021/04/10//####/######.bin",
"method":"PUT","expires_time":1618049385,"ok":true,
"pwd":"changes every time"},
"ok":true}}`

Reply:
Yeah `_sync.gen_presigned_url` appears to match dreame. I haven't really checked for `gen_tmp_presigned_url` so I left it as it is since it's working. I'll see.

I was noticing some issues where it sometimes doesn't update the map with new obstacle info, I was investigating whether the map URL payload affects it (it doesn't)

Reply:
Just checked roborock again on a cloud-provisioned S5 Max

vac → cloud: `{"id":1234,"method":"_sync.batch_gen_room_up_url","params":{"indexes":[12,13,14,15]}}`
cloud → vac: 
```json
{
	"id": 1234,
	"result": [
		"https://awsusor0.fds.api.xiaomi.com/robomap/rubyslite%2F###DID###%2F12?Expires=1618071662000&GalaxyAccessKeyId=#######&Signature=#######",
		"https://awsusor0.fds.api.xiaomi.com/robomap/rubyslite%2F###DID###%2F13?Expires=1618071662000&GalaxyAccessKeyId=#######&Signature=#######",
		"https://awsusor0.fds.api.xiaomi.com/robomap/rubyslite%2F###DID###%2F14?Expires=1618071662000&GalaxyAccessKeyId=#######&Signature=#######",
		"https://awsusor0.fds.api.xiaomi.com/robomap/rubyslite%2F###DID###%2F15?Expires=1618071662000&GalaxyAccessKeyId=#######&Signature=#######"
	]
}
```

Reply:
I don't see any escapes.

Anyway, viomi doesn't seem to use this `batch_gen_room_up_url` thing, but I guess we can definitely try to move the map upload to `MiioValetudoRobot` and see what breaks.